### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by andykuszyk**
* Fixes from https://github.com/form3tech-oss/f1/pull/12
* Additional updates inspired by https://github.com/form3tech/k6-tests/pull/182

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,14 +14,42 @@ import (
...
+
+type MultiStageSetupFn func(t *T) ([]Stage, TeardownFn)
+
+type Stage struct {
+    Name  string
+    RunFn RunFn
+}
+```
+
+Each of these are invoked with the testing support struct shown above. This provides integration with Testify's assert and require capabilities, for making assertions during the tests. This allow clean implementation of scenario test code. It also has a logger pre-configured with markers for the specific test iteration.
```
Comment (markhowardform3) [inline README.md:50]: We're not showing the T struct above.

File: README.md [hunk 2]
```diff
@@ -11,14 +14,42 @@ import (
...
+
+// TeardownFn clears down any resources from a test run after all iterations complete.
+type TeardownFn RunFn
+
+type MultiStageSetupFn func(t *T) ([]Stage, TeardownFn)
+
+type Stage struct {
+    Name  string
+    RunFn RunFn
+}
```
Comment (markhowardform3) [inline README.md:47]: Most people won't use this - I'd leave it out of the top readme.
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -49,17 +80,27 @@ function setupMySuperFastLoadTest(t *testing.T) (testing.RunFn, testing.Teardown
...
 
-## Test lifecyle
-`f1` assumes a lifecyle of tests which is implemented in the `pkg/loadtester` package which basically consists of:
+### Running load tests
+Once you have written a load test and compiled a binary test runner, you can use the various ["trigger modes"](https://github.com/form3tech-oss/f1/tree/master/pkg/f1/trigger) that `f1` supports. These are available as subcommands to the `run` command, so try running `f1 run --help` for more information). The trigger modes currently implemented are as follows:
+
+* `constant` - applies load at a constant rate (e.g. one request per second, irrespective of request duration).
+* `staged` - applies load at various stages (e.g. one request per second for 10s, then two per second for 10s).
+* `users` - applies load from a pool of virtual users (e.g. requests from two virtual users being sent sequentially - they are asa fast or as slow as the requests themselves).
```
Comment (markhowardform3) [inline README.md:88]:
```suggestion
* `users` - applies load from a pool of virtual users (e.g. requests from two virtual users being sent sequentially - they are as fast or as slow as the requests themselves).
```

File: README.md (hunk 1 above)
Comment (andykuszyk) [reply README.md]: Good point, I'm not sure this is necessary either - see 3c4e8796610f061d34f8e4b893c4a536821befc3

File: README.md (hunk 2 above)
Comment (andykuszyk) [reply README.md]: Fixed in 3c4e8796610f061d34f8e4b893c4a536821befc3

**Comment by markhowardform3** [comment]
Maybe we need a picture or two? or possibly https://asciinema.org/ cast of f1 chart & a quick f1 run?

**Pull request merged**